CorsixTH/Lua/research_department.lua

```diff
@@ -499,6 +489,23 @@ function ResearchDepartment:improveMachine(machine)
   end
 end
 
+--[[ Updates the dynamic info for each matching machine found in the player's
+hospital rooms.
+!param machine The machine for determining matches.
+--]]
+function ResearchDepartment:updateMachinesDynamicInfo(machine)
```

Comment:
The parameter `machine` seems to suggest that you will change only one machine (namely the one you pass in), while in fact you change all machines of a given type (eg all inflation machines in the hospital).

To make this more clear, instead of passing in `machine` pass in `machine.id` in the call. Also change the parameter name to something ending in `id` (that is the core data on which you make decisions in the code). For example `machine_id`. Line 500 simply uses the name name (eg `machine_id`). Line 494 also changes a bit.

Reply:
Done.